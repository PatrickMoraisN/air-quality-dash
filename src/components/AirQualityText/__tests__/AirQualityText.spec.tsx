import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import { AirQualityText } from '..'

describe('AirQualityText Component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<AirQualityText airQuality="moderado" />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders the correct air quality text', () => {
    render(<AirQualityText airQuality="bom" />)

    const textElement = screen.getByText('bom')
    expect(textElement).toBeInTheDocument()
  })

  it('renders correctly with the "popup" variant', () => {
    render(<AirQualityText airQuality="péssimo" variant="popup" />)

    const textElement = screen.getByText('péssimo')
    expect(textElement).toBeInTheDocument()

    expect(textElement).toHaveAttribute('variant', 'popup')
  })
})
