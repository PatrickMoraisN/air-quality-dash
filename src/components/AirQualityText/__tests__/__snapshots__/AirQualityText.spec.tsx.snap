// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AirQualityText Component matches snapshot 1`] = `
<DocumentFragment>
  <span
    class="sc-VHjGu ubvvV"
  >
    moderado
  </span>
</DocumentFragment>
`;
